prompt2model/param_selector/search_with_optuna.py

```diff
@@ -0,0 +1,188 @@
+"""This module provides a dummy trainer for testing purposes."""
+from __future__ import annotations  # noqa FI58
+
+from typing import Any
+
+import transformers
+from optuna.trial import Trial
+
+from datasets import Dataset
+from transformers import PreTrainedModel  # noqa
+from transformers import PreTrainedTokenizer
+from prompt2model.prompt_parser.base import PromptSpec
+
+from prompt2model.utils.config import DEFAULT_HYPERPARAMETERS, DEFAULT_HYPERPARAMETERS_SPACE
+from prompt2model.model_trainer import BaseTrainer
+from prompt2model.param_selector.base import ParamSelector
+
+# TODO:
+# - User tweaking hyperparameter
+# - Dynamic initialization of hyperparamter range from task type and complexity
+# - Using LLM to suggest hyperparameter
```

Comment:
Let's not keep TODOs in the code. Instead, please create issues to track these suggestions 😄

Reply:
Sure, will be putting issues from the next set of commits